activerecord/test/cases/connection_adapters/connection_handler_test.rb

```diff
@@ -406,33 +406,33 @@ def test_retrieve_connection_pool_copies_schema_cache_from_ancestor_pool
         def test_pool_from_any_process_for_uses_most_recent_spec
           skip unless current_adapter?(:SQLite3Adapter)
 
-          file = Tempfile.new "lol.sqlite3"
+          begin
+            file = Tempfile.new "lol.sqlite3"
 
-          rd, wr = IO.pipe
-          rd.binmode
-          wr.binmode
+            rd, wr = IO.pipe
+            rd.binmode
+            wr.binmode
 
-          pid = fork do
-            config_hash = ActiveRecord::Base.configurations.configs_for(env_name: "arunit", name: "primary").configuration_hash.merge(database: file.path)
-            ActiveRecord::Base.establish_connection(config_hash)
+            pid = fork do
+              config_hash = ActiveRecord::Base.configurations.configs_for(env_name: "arunit", name: "primary").configuration_hash.merge(database: file.path)
+              ActiveRecord::Base.establish_connection(config_hash)
 
-            pid2 = fork do
-              wr.write ActiveRecord::Base.connection_db_config.database
-              wr.close
-            end
+              pid2 = fork do
+                wr.write ActiveRecord::Base.connection_db_config.database
+                wr.close
+              end
 
-            Process.waitpid pid2
-          end
+              Process.waitpid pid2
+            end
 
-          Process.waitpid pid
+            Process.waitpid pid
 
-          wr.close
+            wr.close
 
-          assert_equal file.path, rd.read
+            assert_equal file.path, rd.read
 
-          rd.close
-        ensure
-          if file
```

Comment:
Hmm... as well as disliking that this makes the test more indented, I disagree that `if file` is redundant: if `Tempfile.new` raises, this will have a bad time.